src/mpcCoreKit.ts

```diff
@@ -228,40 +208,61 @@ export class Web3AuthMPCCoreKit implements ICoreKit, IMPCContext {
+  // }
+
+  public getSupportedSigTypes(): Array<SigType> {
+    return Array.from(this.supportedSigTypes);
   }
 
-  public setCustomSessionSigGenerator(sessionSigGenerator: ISessionSigGenerator) {
-    this.sessionSigGenerator = sessionSigGenerator;
+  public getSupportedCurveKeyTypes(): Array<KeyType> {
+    return Array.from(this.supportedCurveKeyTypes);
   }
 
-  async getSessionSignatures(): Promise<string[]> {
-    return this.sessionSigGenerator.getSessionSigs();
+  /**
+   * The threshold signing library to use.
+   */
+  public addTssLibs(tssLibs: TssLibType[], force = false) {
+    for (const tssLib of tssLibs) {
+      if (!this.supportedCurveKeyTypes.has(tssLib.keyType as KeyType)) {
+        throw CoreKitError.invalidConfig(
+          `Tsslib key type: ${tssLib.keyType} is not supported, please reconfigure corekit with supported curves keytype`
+        );
+      }
+      if (this.supportedSigTypes.has(tssLib.sigType as WEB3AUTH_SIG_TYPE)) {
+        if (force) {
+          this._tssLibs = this._tssLibs.filter((t) => t.sigType !== tssLib.sigType);
+          this._tssLibs.push(tssLib);
+        } else {
+          log.warn(`Tsslib sig type: ${tssLib.sigType} is already exists, library is not appended`);
+        }
+        continue;
+      }
+      this.supportedSigTypes.add(tssLib.sigType as WEB3AUTH_SIG_TYPE);
+      this._tssLibs.push(tssLib);
+    }
   }
 
-  public setTkeyType(tkeyType: KeyType) {
-    // check tkeyType is supported by tssLib
-    if (!this.supportedCurveKeyTypes.has(tkeyType)) throw CoreError.default("KeyType not supported, please provide valid tssLib");
-    this._keyType = tkeyType;
+  public setCustomSessionSigGenerator(sessionSigGenerator: ISessionSigGenerator) {
+    this.sessionSigGenerator = sessionSigGenerator;
   }
 
-  public setSigType(sigType: WEB3AUTH_SIG_TYPE) {
-    // check tkeyType is supported by tssLib
-    if (!this.supportedSigTypes.has(sigType)) throw CoreError.default("SigType not supported, please provide valid tssLib");
-    this._sigType = sigType;
+  async getSessionSignatures(): Promise<string[]> {
+    return this.sessionSigGenerator.getSessionSigs();
   }
 
-  public getTssShare(factorkey: BN, accountIndex?: number) {
+  public getTssShare(args: { keyType: KeyType; factorkey: BN; accountIndex?: number }) {
+    const { keyType, factorkey, accountIndex } = args;
     return this.tkey.getTSSShare(factorkey, {
       tssTag: TSS_TAG_DEFAULT,
-      keyType: this.keyType,
+      keyType,
       accountIndex: accountIndex === undefined ? this.state.accountIndex : accountIndex,
     });
   }
 
-  public getTssData(args: { skipThrow: boolean; keyType?: KeyType } = { skipThrow: false }) {
-    const result = this.tkey.metadata.getTssData(args.keyType ?? this.keyType, TSS_TAG_DEFAULT);
+  public getTssData(args: { skipThrow?: boolean; keyType: KeyType }) {
```

Comment:
this `skipThrow` flag looks super odd. if you want to skip an error, just catch and discard!?

Reply:
break the function to 2 functions for more clearity